src/Avalonia.Controls/Platform/PlatformManager.cs

```diff
@@ -21,26 +21,19 @@ public static void SetDesignerScalingFactor(double factor)
         }
 
         public static ITrayIconImpl? CreateTrayIcon() =>
-            s_designerMode ? null : AvaloniaLocator.Current.GetService<IWindowingPlatform>()?.CreateTrayIcon();
+            s_designerMode ? null : AvaloniaLocator.Current.GetRequiredService<IWindowingPlatform>().CreateTrayIcon();
```

Comment:
```suggestion
            s_designerMode ? null : AvaloniaLocator.Current.GetService<IWindowingPlatform>()?.CreateTrayIcon();
```

CreateTrayIcon should return null without any IWindowingPlatform registered. At least at the moment that's what framework expects having window-less backends.